custom/icds/models.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.app_manager.dbaccessors import get_build_by_version
+from corehq.motech.utils import b64_aes_decrypt
+from custom.icds.utils.ccz_hosting import CCZHostingUtility
+from custom.icds.validators import (
+    LowercaseAlphanumbericValidator,
+)
+
+
+class CCZHostingLink(models.Model):
+    identifier = models.CharField(null=False, unique=True, max_length=255, db_index=True,
+                                  validators=[LowercaseAlphanumbericValidator])
+    username = models.CharField(null=False, max_length=255)
+    # b64_aes_encrypt'ed raw password is stored in DB
+    password = models.CharField(null=False, max_length=255)
+    domain = models.CharField(null=False, max_length=255)
+    page_title = models.CharField(blank=True, max_length=255)
+
+    def __str__(self):
+        return self.identifier
+
+    @cached_property
+    def get_password(self):
+        return b64_aes_decrypt(self.password)
+
+    def to_json(self):
+        from custom.icds.serializers import CCZHostingLinkSerializer
+        return CCZHostingLinkSerializer(self).data
+
+    def delete(self, *args, **kwargs):
+        for ccz_hosting in CCZHosting.objects.filter(link=self):
+            ccz_hosting.delete()
+        super(CCZHostingLink, self).delete(*args, **kwargs)
+
+
+class CCZHosting(models.Model):
+    link = models.ForeignKey(CCZHostingLink, on_delete=models.CASCADE)
+    app_id = models.CharField(max_length=255, null=False)
+    version = models.IntegerField(null=False)
+    profile_id = models.CharField(max_length=255, blank=True)
+    file_name = models.CharField(max_length=255, blank=True)
+
+    class Meta:
+        unique_together = ('link', 'app_id', 'version', 'profile_id')
```

Comment:
Why would you have multiple links for the same app + version + build profile combination?

Reply:
To phrase another way, why does `link` need to be in this list, shouldn't the other three alone be unique?

Reply:
@orangejenny 
We can have the same version and profile on an app on two links like two states using the same language/profile and on the same version of the app.